torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -509,6 +509,14 @@ def _root_pre_forward(
     args = args_tuple[0]
     kwargs = kwargs_tuple[0]
 
+    input_dtype: Optional[torch.dtype] = state.mixed_precision.param_dtype
+    # Cast forward inputs in root FSDP instance if cast_root_foward_inputs is set as True
```

Comment:
nit: I think we can simplify the comment.

The fact that `cast_forward_inputs == True` implies that we cast forward inputs on the root FSDP is clear from L516-519.

I think the two non-obvious points to capture are:
1. As you mentioned already in the latter half of the comment, if `not cast_root_forward_inputs`, then there is still a chance to cast forward inputs per FSDP instance based on their individual `cast_forward_inputs` values.
2. For the root FSDP, the `cast_root_forward_inputs` value takes precedence over its `cast_forward_inputs` value. Given the default settings (`cast_forward_inputs=False`, `cast_root_forward_inputs=True`), we _want_ this override behavior. For non-root FSDP, the `cast_root_forward_inputs` value is actually ignored.

I feel that these two may actually useful to share with the user as well in public documentation (e.g. as a note in the `MixedPrecision` docstring in `api.py`) rather than putting it here inline.